### PR TITLE
AppsController split: `kill task` method

### PR DESCRIPTION
**Pull request by meln1k**
Summary: splitting a big appsController pr, this time `kill task` method

JIRA issues: MARATHON-7299

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala [hunk 1]
```diff
@@ -76,4 +76,30 @@ object Directives extends AuthDirectives with LeaderDirectives with AkkaDirectiv
       case Success => f(Unit)
     }
   }
+
+  def extractTaskKillingMode: Directive1[TaskKillingMode] =
+    (parameter("scale".as[Boolean].?(false)) & parameter("wipe".as[Boolean].?(false))).tflatMap {
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala]: What is `tflatMap`?

**Comment by meln1k** [comment]
@jeschkies no, it doesn't, although they both contain new directive.

File: src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala (hunk 1 above)
Comment (meln1k) [reply src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala]: A something like `flatMap` but for directives :)

Comment (alenkacz) [reply src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala]: I did not know that one either. Thanks for asking and answering :)

**Comment by alenkacz** [comment]
@meln1k now that I look at it the other day with clear head ... why don't we need any kind of authorization for killing tasks? For listing tasks it is present...

File: src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala (hunk 1 above)
Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala]: I actually stumbled upon this as well.

File: src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala [hunk 2]
```diff
@@ -98,6 +101,31 @@ object Directives extends AuthDirectives with LeaderDirectives with AkkaDirectiv
     }
   }
 
+  def extractTaskKillingMode: Directive1[TaskKillingMode] =
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala]: How is this different from `AppsDirectives.extractTaskKillingMode`?

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala [hunk 3]
```diff
@@ -326,7 +326,36 @@ class AppsController(
 
   private def killTasks(appId: PathId)(implicit identity: Identity): Route = ???
 
-  private def killTask(appId: PathId, taskId: TaskId)(implicit identity: Identity): Route = ???
+  private def killTask(appId: PathId, taskId: TaskId)(implicit identity: Identity): Route = {
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala]: This looks incredibly similar to `killTasks` from #5679. The only difference seems to be `findToKill`. What do you think about consolidating these two diffs? You could make one branch the base of the other.

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala [hunk 4]
```diff
@@ -2343,5 +2352,60 @@ class AppsControllerTest extends UnitTest with GroupCreation with ScalatestRoute
...
     }
 
+    "Kill task and scale" in new Fixture {
+      val app = AppDefinition(id = PathId("/app"))
+      val rootGroup = createRootGroup(Map(app.id -> app))
+      val plan = DeploymentPlan(rootGroup, rootGroup)
+      groupManager.app(PathId("/app")) returns Some(app)
+
+      groupManager.rootGroup() returns rootGroup
+      taskKiller.killAndScale(any, any, any)(any) returns Future.successful(plan)
```
Comment (jeschkies) [inline src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala]: Hm, shouldn't we match the arguments?

Comment (meln1k) [reply src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala]: Good catch!

**Pull request merged**